### DISCUSSION TITLE
guides/model-training-tips/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#14842] [Docs]
# guides/model-training-tips/

Find best practices, optimization strategies, and troubleshooting advice for training computer vision models. Improve your model training efficiency and accuracy.

https://docs.ultralytics.com/guides/model-training-tips/

**Comment by Taufiq-Ahmed** [comment]
Hi I have a question. I am using Yolov8s for trained a dataset that has four class.  fire, smoke, person, animal.  all the images has areal view to detect forest fire . In my dataset I have forest fire images from drone so after training it detect fire smoke quite accurately. however, I do not have any dataset that has person or animal in forest fire. so I have to collect some other dataset from roboflow  that has person and animal and the n I have merged them with my own dataset ( have changed the label of animal and person accordingly so 0 means smoke, 1 means fire, 2 means person and 3 means animal and also updated the data.yaml file) . this time the problem is it could not detect the animal and person properly some times it can detect them. The main reasons seems to me is that the drone is flying in a high altitude so it does not detect person and animal properly because in the dataset of person and animal the view of the drone is not that much high. is there any technique to solve this problem and detect all the objects as many as possible?

**Reply by glenn-jocher** [reply]
@Taufiq-Ahmed it sounds like the altitude difference between your training and test datasets is causing the detection issues. To address this, consider using data augmentation techniques to simulate higher altitude views in your training data. You can also try multiscale training to help the model generalize better across different scales. Additionally, fine-tuning the model with a mix of high-altitude and low-altitude images might improve its performance. For more detailed strategies, you can refer to our [model training tips](https://docs.ultralytics.com/guides/model-training-tips/).

**Comment by Imran-ml** [comment]
First of all its great to have another powerful model and congratulations. I have fine tuned and trained yolo11x.pt for my custom objects but I'm facing some issues. My images data resolution is around 6000 x 2000 but while trained the model I set the imgsz=(1280, 640). It work well when I adjust the video stream by zoom in-out and detect all my 60 objects well (objects are small) but when there is some zoom or squeezed then it completely dont detect any object or few only.

Please let me know if I'm missing something or I have to do something else?
I have labelled the data on my original images so resizing making any issues? any ratio issues? Please guide me properly so I can use this power yolo11 model effectively.

**Reply by glenn-jocher** [reply]
@Imran-ml it sounds like the issue might be related to image resizing affecting object detection. Consider enabling multiscale training by adjusting the `scale` parameter to help the model generalize better across different image sizes. This can improve detection performance when objects appear at varying scales. For more detailed guidance, you can refer to our model training tips at https://docs.ultralytics.com/guides/model-training-tips/.

**Comment by VeeranjaneyuluToka** [comment]
Hi, I just noticed a thing, looks like the optimizer is getting chosen based on no.of iterations, would you mind telling me the reason for the same? I mean how does iterations matter when we choose optimizer?

**Comment by TSHSLU** [comment]
What is the minimum Dataset Size required if i use a pretrained model for custom projects. i barely find any sources online which recommend a minimum dataset size.

**Reply by glenn-jocher** [reply]
There isn’t a fixed “minimum” dataset size when fine-tuning a pretrained model; if your domain is close to the pretrained data and labels are clean, you can often start getting useful signal with ~50–200 images per class (or a few hundred labeled object instances), but for reliably robust results across different lighting/viewpoints/backgrounds you’ll usually want 500–2000+ images per class with good diversity—one practical way to find your true minimum is to train on a small pilot subset, check validation failures, then add data specifically where the model breaks (the `fraction` workflow in the [Model training tips guide](https://docs.ultralytics.com/guides/model-training-tips/) helps iterate quickly).

```python
from ultralytics import YOLO

model = YOLO("yolo26n.pt")
model.train(data="data.yaml", epochs=100, fraction=0.1)
```

**Comment by puzzlemindgames** [comment]
Hello

I'm working on a browser-based puzzle solver that uses a custom YOLO model to detect visible objects from game screenshots.

Some examples include:
Transparent ice/frost overlays on the lower part of objects
Semi-transparent cloth or curtain overlays
Small decorative objects inside larger objects (for example, keys inside containers)
The visible objects are usually detected correctly, but these overlays often cause false negatives or incorrect class predictions.

At the moment I'm considering a few approaches:
Training with many more occlusion-heavy examples
Splitting the detection into multiple stages instead of one larger model

**python**
results = model.predict(
    source="level_512.png",
    conf=0.25,
    iou=0.45,
    imgsz=1280
)
for box in results[0].boxes:
    print(box.cls, box.conf)

For anyone who has trained YOLO models on UI or game screenshots which approach gave you the biggest improvement? 
Or is there another strategy that tends to work better for partially occluded objects?

I'd really appreciate hearing about real-world experiences before I spend time retraining the dataset.

**Reply by glenn-jocher** [reply]
Start with targeted data: add representative occluded examples, keep annotations consistent for partially visible objects, and synthetically composite the actual frost/cloth overlays at realistic opacity and placement. Measure recall per class on a dedicated occlusion-heavy validation split, then tune [`mosaic`, `mixup`, or `cutmix`](https://docs.ultralytics.com/guides/yolo-data-augmentation) conservatively; higher `imgsz` or tiled inference can help nested small objects. A multi-stage pipeline is usually worthwhile only if the first stage can reliably isolate containers or overlay regions. Please continue sharing results and questions here so the Ultralytics team and community can see them.

**Reply by puzzlemindgames** [reply]
Thanks! I followed your suggestions and added more occluded samples. The main objects/tubes are much more stable as well as the colors now but I'm still struggling with very small objects.

i cant able to attach model training screenshots here for you to review.

For example, tiny keys or tags inside transparent containers are often missed or detected as the wrong class.
For context the screenshots come from a puzzle game with objects similar to these shown in sample pictures embedded at product homepage:
https://aimagicsortsolver.com/

results = model.predict(
    source="level.png",
    imgsz=1280,
    conf=0.35,
    iou=0.45
)
Would you recommend keep improving a single model or train a second YOLO model just for these small objects?

**Reply by glenn-jocher** [reply]
Keep the single model first: audit tiny-object annotations, add more balanced close-up examples, and test cropped or [SAHI tiled inference](https://docs.ultralytics.com/guides/sahi-tiled-inference) so keys and tags occupy more pixels; also compare `conf=0.15–0.25`, since `0.35` may suppress weak detections. Train a second specialized model only if the first model reliably detects each container, allowing its crop to be passed to the small-object model at higher effective resolution.

**Reply by puzzlemindgames** [reply]
Thank you for your suggestions.
I'll tweak it as per the recommendations.
